components/bitcoin-topics/markdown-ui/opcode-explorer.tsx

```diff
@@ -0,0 +1,449 @@
+                const duration = playerRef.current?.duration || 0
+                const newStep = Math.floor(currentTime / 1000)
+                if (newStep !== currentStep) {
+                    setCurrentStep(newStep)
+                }
+                if (currentTime >= duration) {
+                    setIsPlaying(false)
+                    setCurrentStep(0)
+                    playerRef.current?.seekTo(0)
+                    clearInterval(interval)
+                }
+            }, 100)
+            return () => clearInterval(interval)
+        }
+    }, [isPlaying, currentStep, totalSteps])
+
+    const playOneSecond = useCallback((startTime: number) => {
+        if (playerRef.current) {
+            if (animationTimeoutRef.current) {
+                clearTimeout(animationTimeoutRef.current)
+            }
+            setIsAnimating(true)
+            setIsPlaying(true)
+            playerRef.current.seekTo(startTime)
+            playerRef.current.play()
+            animationTimeoutRef.current = setTimeout(() => {
+                playerRef.current?.pause()
+                setIsAnimating(false)
+                setIsPlaying(false)
+                setHasStarted(true)
+            }, 1000)
+        }
+    }, [])
+
+    const handlePrevious = useCallback(() => {
+        if (isAnimating) return
+        setCurrentStep((prev) => {
+            const newStep = Math.max(0, prev - 1)
+            playOneSecond(newStep * 1000)
+            return newStep
+        })
+    }, [isAnimating, playOneSecond])
+
+    const handleNext = useCallback(() => {
+        if (isAnimating) return
+        if (!hasStarted) {
+            playOneSecond(0)
+            setCurrentStep(0)
+        } else {
+            setCurrentStep((prev) => {
+                const newStep = Math.min(totalSteps - 1, prev + 1)
+                playOneSecond(newStep * 1000)
+                return newStep
+            })
+        }
+    }, [isAnimating, playOneSecond, totalSteps, hasStarted])
+
+    const handlePlay = useCallback(() => {
+        if (playerRef.current) {
+            if (isAnimating) {
+                playerRef.current.play()
+            } else {
+                playOneSecond(currentStep * 1000)
+            }
+            setIsPlaying(true)
+        }
+    }, [currentStep, isAnimating, playOneSecond])
+
+    const handlePause = useCallback(() => {
+        if (playerRef.current) {
+            playerRef.current.pause()
+            setIsPlaying(false)
+            if (animationTimeoutRef.current) {
+                clearTimeout(animationTimeoutRef.current)
+            }
+            setIsAnimating(false)
+        }
+    }, [])
+
+    useEffect(() => {
+        return () => {
+            if (animationTimeoutRef.current) {
+                clearTimeout(animationTimeoutRef.current)
+            }
+        }
+    }, [])
+
+    return (
+        <div className="mx-auto py-1 -mx-4 sm:-mx-8 md:-mx-16 lg:-mx-32 xl:-mx-40">
```

Comment:
```suggestion
        <div className="mx-auto py-1">
```
negative margins breaks screen layout